### PR TITLE
Add envelope algorithm

### PR DESCRIPTION
**Pull request by jfirebaugh**

### CODE REVIEW

File: include/mapbox/geometry/box.hpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+struct box
+{
+    using point_type = point<T>;
+
+    box(point_type const& min, point_type const& max)
+        : min_corner(min), max_corner(max)
+    {}
+
+    point_type min_corner;
+    point_type max_corner;
```
Comment (mourner) [inline include/mapbox/geometry/box.hpp]: Maybe just `min`/`max`?

Comment (artemp) [inline include/mapbox/geometry/box.hpp]: +1 min/max !

**Comment by jfirebaugh** [comment]
@springmeyer noted that a potential optimization is to ignore polygon interior rings, which produces the same output for all valid geometry. It changes the results for polygons whose interior rings aren't actually interior, obviously. Not sure I have a good sense of whether that matters yet.

**Comment by springmeyer** [comment]
> Not sure I have a good sense of whether that matters yet.

My opinion is to not account for interior rings in the envelope. In my usage/experience if it did change the results that would be bad - the case I'm thinking of is filtering out geometries that don't intersect with a tile. An interior ring without a polygon that did intersect with a boundary when its exterior did not would be a great candidate to get thrown out.

**Comment by mourner** [comment]
Let's merge this as is and release a version — I need the code for geojson-vt-cpp. We can do the omit-holes optimization later.

**Pull request merged**